### DISCUSSION TITLE
create-react-app using webpack 4

### DISCUSSION THREAD

**Discussion by kiehnek** [panva/jose#133] [Q&A] [answered]
I created a basic React App with Typescript using `create-react-app`. (And installed jose)
When I try to import something from `jose/something/something`the Compiler shows the following Error: `Could not find module in path: 'jose/util/generate_secret' relative to '/src/App.tsx'`.

I think this is some strange way Webpack behaves with the config options of `create-react-app`.

I also just wanted to thank you for this great library. It's my go-to destination for encryption/signing/JWT ...

**Comment by kiehnek** [comment]
Minimal Example: https://codesandbox.io/s/crimson-darkness-35vtm?file=/src/App.tsx

**Comment by panva** [comment, marked as answer]
webpack 4 does not support [subpath exports](https://nodejs.org/api/packages.html#packages_subpath_exports). [webpack 5 has no issue](https://github.com/panva/jose/issues/111)

**Reply by panva** [reply]
create-react-app support for webpack 5 is [on the way](https://github.com/facebook/create-react-app/issues/9994)

**Reply by yourfavorite** [reply]
Is there a workaround to get jose v3 working with webpack 4?

**Comment by panva** [comment]
For the record this is not a problem with react, nor jose. Webpack 4 used by create-react-app does not support a required module resolution feature, webpack 5 does.

**Comment by kiehnek** [comment]
Thanks for the quick answer! I thougt create-react-app is already using Webpack 5 but apparently I was wrong.

**Comment by maximivanov** [comment]
I can confirm jose works perfectly with webpack 5. I've used this bolierplate for a quick test: https://github.com/rwieruch/minimal-react-webpack-babel-setup

**Reply by kiehnek** [reply]
Yes, I used jose before in a plain typescript project with webpack 5 and it worked perfectly.

**Comment by panva** [comment]
I'm done waiting for these ecosystems to catch up. jose v4.x will be coming out today/tomorrow with only top level named exports that solve all these tools' problems.
